optuna/visualization/_contour.py

```diff
@@ -129,7 +129,7 @@ def _get_contour_plot(
 
         elif _is_categorical(trials, p_name):
             # For numeric values, plotly does not automatically plot as "category" type.
-            update_category_axes[p_name] = any([str(v).isnumeric() for v in set(values)])
+            update_category_axes[p_name] = any([_is_numeric(str(v)) for v in set(values)])
```

Comment:
A nit-picky, but `[]` is not necessary for `any()`.
This is not introduced by this PR, but could you fix it to keep the code clean?

```suggestion
            update_category_axes[p_name] = any(_is_numeric(str(v)) for v in set(values))
```

Reply:
Make code cleaner